File: lachaudiere/src/application_core/domain/entities/Categorie.php
<?php
namespace lachaudiere\application_core\domain\entities;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Categorie extends Model {
    protected $table = 'categories';
    protected $primaryKey = 'id_categorie';
    public $timestamps = false;

    protected $fillable = ['libelle', 'description'];

    public function evenements(): HasMany {
        return $this->hasMany(Evenement::class, 'id_categorie');
    }
}

File: lachaudiere_app/lib/models/categorie.dart
class Categorie {
  final int id;
  final String libelle;
  final String? description;

  Categorie({required this.id, required this.libelle, this.description});

  factory Categorie.fromJson(Map<String, dynamic> json) {
    return Categorie(
      id: json['id_categorie'] as int? ?? 0,
      libelle: json['libelle'] as String? ?? 'Non classé',
      description: json['description'] as String?,
    );
  }
}